### ISSUE TITLE
Where to find go2rtc.yaml when running WebRTC in a home assistant docker container?

### ISSUE THREAD

**Issue by cameronjust** [AlexxIT/WebRTC#589]
Apologies if this is the wrong place to ask but I couldn't find a forum anywhere to ask.

**##################**
**# Current State**

Camera Setup is 6 cameras connected to an Amcrest NVR.

Docker Containers on Unraid all referencing the rtsp and snapshots APIs on the Amcreat NVR

 * Home Assistant - Running in Host mode with HACS installed and WebRTC for dashboards
 * Frigate

**##############**
**# Desired State**

- Get rid of the Amcrest NVR and have all cameras configured with Go2RTC.
- Record all cameras 24/7 for up to 2 weeks
- Record all cameras motion only for up to 3 months
- Frigate object detection
- Dashboards in Home Assistant to view live camera feeds

**Proposed Docker Containers**

- Go2RTC - To act as a proxy for all cameras.  All docker containers below will reference cameras via gotortc.
- Home Assistant - For viewing cameras on dashboards
- Frigate - For image detection and to keep motion only recordings for up to 3 months
- Shinobi (or equivalent) - To keep a 24/7 recording of all cameras for at least 2 weeks. Frigate only lets you keep motion only OR 24/7 recordings not both at different retention levels
 
**############**
**# Questions**

I'm leaning towards using Go2RTC as a seperate container so that is the only app that connects to the cameras. Home Assistant, Frigate and Shinobi will pull data feeds from Go2RTC so only one stream is coming from the cameras and these docker containers will be pulling their feeds from Go2RTC on the same host. Less traffic on the network and less effort by the cameras sending only one stream to Go2RTC docker.

WebRTC installed on Home Assistant uses up the ports that Go2RTC container would use. How do I disable this in Home Assistant/WebRTC?

I can see the documentation for WebRTC says "You can change the go2rtc settings by adding the go2rtc.yaml file to your Hass configuration folder."
Ref: https://github.com/AlexxIT/WebRTC#go2rtc

 For the life of me I can't find go2rtc.yaml  anywhere in /mnt/cache/appdata/homeassistant?

Do I even need WebRTC in home assistant if I have Go2RTC as a seperate docker container? Sample of one of my dashboard panels in home assistant

```
type: custom:webrtc-camera
entity: camera.front
style: 'video {aspect-ratio: 16/9; object-fit: fill;}'
show_header_toggle: false
view_layout:
  grid-area: front

```

Am I going about this the wrong way and is there a better way to achieve this?

**Comment by AlexxIT** [comment]
You don't have a go2rtc.yaml because you haven't created one.

**Comment by cameronjust** [comment]
Thanks for the reply.

Doh. Well that's an obvious mistake isn't it?

OK I tried to create go2rtc.yaml to disable go2rtc completely from home assistant container so I can use a dedicated docker container for go2rtc.

It didn't seem to work though?

**/mnt/cache/appdata/homeassistant/go2rtc.yaml** 
```
# netstat -ln | grep '8554\|1984\|8555'

# defaults to 1984
api:
  listen: ""

# defaults to 8554
rtsp:
  listen: ""

# Defaults to 8555
webrtc:
  listen: ""  
```
I then restarted the home assistant container and go2rtc was still running?

```
root@Tower:/mnt/cache/appdata/homeassistant# ps ax | grep go2rtc
 2949 pts/2    S+     0:00 grep go2rtc
26100 ?        Sl     0:00 /config/go2rtc-1.8.0
27476 ?        S      0:00 s6-supervise go2rtc
27478 ?        S      0:00 s6-supervise go2rtc-healthcheck
27479 ?        S      0:00 s6-supervise go2rtc-log
27541 ?        Ss     0:00 s6-log -b -- T 1 n0 s10000000 T /dev/shm/logs/go2rtc
27546 ?        Ssl    0:00 /usr/local/go2rtc/bin/go2rtc -config=/dev/shm/go2rtc.yaml
27570 ?        Ss     0:00 bash ./run.user go2rtc-healthcheck
```

and the ports were still active

```
root@Tower:/mnt/cache/appdata/homeassistant# netstat -ln | grep '8554\|1984\|8555'
tcp        0      0 0.0.0.0:8555            0.0.0.0:*               LISTEN     
tcp6       0      0 :::1984                 :::*                    LISTEN     
tcp6       0      0 :::8554                 :::*                    LISTEN     

```

**Comment by AlexxIT** [comment]
Probably it's go2rtc inside Frigate. You can't stop it. You should change it settings inside Frigate config.

**Comment by cameronjust** [comment]
It seems to be coming from the Home Assistant Docker and not Frigate. Just did some tests.

**With both Frigate and Home Assistant Dockers running**
```
root@Tower:/mnt/cache/appdata/homeassistant# netstat -ln | grep '8554\|1984\|8555'
tcp        0      0 0.0.0.0:8555            0.0.0.0:*               LISTEN
tcp6       0      0 :::1984                 :::*                    LISTEN
tcp6       0      0 :::8554                 :::*                    LISTEN
root@Tower:/mnt/cache/appdata/homeassistant#
```

**Stopped Frigate** - Still running

```
root@Tower:/mnt/cache/appdata/homeassistant# netstat -ln | grep '8554\|1984\|8555'
tcp        0      0 0.0.0.0:8555            0.0.0.0:*               LISTEN
tcp6       0      0 :::1984                 :::*                    LISTEN
tcp6       0      0 :::8554                 :::*                    LISTEN
root@Tower:/mnt/cache/appdata/homeassistant#
```

**Stopped Home Assistant** - Go2RTC Stopped

```
root@Tower:/mnt/cache/appdata/homeassistant# netstat -ln | grep '8554\|1984\|8555'
root@Tower:/mnt/cache/appdata/homeassistant#
```

**Started Home Assistant with Frigate still stopped** - go2rtc started up

```
root@Tower:/mnt/cache/appdata/homeassistant# netstat -ln | grep '8554\|1984\|8555'
tcp        0      0 0.0.0.0:8555            0.0.0.0:*               LISTEN
tcp6       0      0 :::1984                 :::*                    LISTEN
tcp6       0      0 :::8554                 :::*                    LISTEN
root@Tower:/mnt/cache/appdata/homeassistant#
```

**Comment by cameronjust** [comment]
Apologies for ressurecting this old thread but I've decided to try to get this sorted once and for all.

I've ruled out frigate being the issue and know that the listener for port 1984 is coming from the home assistant HACS installation of "WebRTC Camera" I just can't work out how to get my custom config to load to disable the api listener on port 1984.

I've editted the go2rtc.yaml file in the HASS directory as instructed but it just doesn't seem to load when I start Home Assistant. Looking at the config editor.html for the web service at port 1984 shows that my go2rtc.yaml file was never loaded.

![gotortc-config-issues](https://github.com/AlexxIT/WebRTC/assets/66708090/d499d1e2-ac96-4036-8528-191e3c685e62)

If I can crack this I'll just have go2rtc running as a seperate docker container and I'll configure it that way.

I'm sure it's something simple I'm missing.

**Comment by AlexxIT** [comment]
It's better to use only one go2rtc. So you need to remove WebRTC integration, run Frigate and setup again WebRTC integration with link to Frigate's go2rtc.

**Comment by cameronjust** [comment]
> It's better to use only one go2rtc. So you need to remove WebRTC integration, run Frigate and setup again WebRTC integration with link to Frigate's go2rtc.

That is 100% my plan longer term to just have one go2rtc as a standalone docker container. Home Assistant and Frigate will use that go2rtc.

What's puzzling me though in this interim step is why my Home Assistant WebRTC isn't reading in the go2rtc.yaml file at all. I was poking around in your [__init__.py](https://github.com/AlexxIT/WebRTC/blob/master/custom_components/webrtc/__init__.py) last night to try to write some INFO logging to see where or if it's trying to load my go2rtc.yaml from the hass config dir but couldn't see anything I added around lines 123 of that file. I suspect those logs must be written elsewhere and not in the HASS container logs.

**Comment by cameronjust** [comment]
Finally worked it out. I completely uninstalled the WetRTC integration from Home Assistant then readded it and had the option of selecting my go2rtc docker container.

After tweaking I worked out I didn't need to completely uninstall it I just needed to delete the integration config and re-add it like this.
![gotortc-config-solution](https://github.com/AlexxIT/WebRTC/assets/66708090/61a9aaa1-b281-4671-85dc-7829de4e9af1)
